From c8a9435f96947998544056982670e9fe6f39b0dd Mon Sep 17 00:00:00 2001
From: Steven McDonald <steven@steven-mcdonald.id.au>
Date: Tue, 29 Apr 2014 15:51:22 +1000
Subject: [PATCH] Search the full config database for a question's template

Currently, Debconf::DbDriver only searches the current database
(referred to by $this) for a template, which doesn't always work
correctly when using stacked databases.
---
 Debconf/DbDriver.pm | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Debconf/DbDriver.pm b/Debconf/DbDriver.pm
index ef6bdd6..b0db2fa 100644
--- a/Debconf/DbDriver.pm
+++ b/Debconf/DbDriver.pm
@@ -206,7 +206,7 @@ sub accept {
 
 	if (exists $this->{accept_type} || exists $this->{reject_type}) {
 		if (! defined $type || ! length $type) {
-			my $template = Debconf::Template->get($this->getfield($name, 'template'));
+			my $template = Debconf::Template->get($Debconf::Db::config->getfield($name, 'template'));
 			return 1 unless $template; # no type to act on
 			$type=$template->type || '';
 		}
@@ -228,7 +228,7 @@ sub ispassword {
 	my $this=shift;
 	my $item=shift;
 
-	my $template=$this->getfield($item, 'template');
+	my $template=$Debconf::Db::config->getfield($item, 'template');
 	return unless defined $template;
 	$template=Debconf::Template->get($template);
 	return unless $template;
-- 
2.0.0.rc0

